analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotShiftByZeroOrIntSize.cs

```diff
@@ -168,24 +167,22 @@ private static ShiftInstance FindShiftInstance(SyntaxNode node, SemanticModel se
                 return new ShiftInstance(node);
```

Comment:
This and the previous two return statements are identical, I suggest merging the three ifs.

Reply:
Before your changes, the semantic model was only used if the shift value was a constant, after your last changes we always go to retrieve the type symbol which in theory should have a performance impact. What I had in mind when I suggested merging the three ifs is something like this.

```
            if (!TryGetConstantValue(tuple.Item2, out var shiftByCount)
                || semanticModel.GetTypeInfo(node).ConvertedType is not { } typeSymbol
                || (FindTypeSizeOrDefault(typeSymbol) is var variableBitLength &&  variableBitLength == 0))
```

Reply:
Oh I see, that makes more sense, thanks.